app/controllers/posts_controller.rb

```diff
@@ -13,36 +13,57 @@ def show
 
   def new
     @community = Community.find(params[:community_id])
-    @post = Post.new
+    @post = Post.new 
+    @drafts = Post.order(created_at: :desc).page(params[:page]).per 5
   end
 
   def create
     @post = Post.new post_values
     @post.account_id = current_account.id
     @post.community_id = params[:community_id]
-
     if @post.save
-      redirect_to community_path(@post.community_id)
+      if params[:commit] == "Publish"
+        @post.is_drafted = false
+        @post.save
+        redirect_to community_path(@post.community_id)
+      else 
+        @post.is_drafted = true
+        @post.save
+        redirect_to draft_path
+      end 
     else
       @community = Community.find(params[:community_id])
       render :new
     end
   end
 
   def edit
+    @post = Post.find(params[:id])
     @community = Community.find(params[:community_id])
   end
 
+  def draft 
+    @drafts =  Post.order(created_at: :desc).page(params[:page]).per 7
+  end 
+
 
   def update
-    @community = Community.find(params[:community_id])
-    if @community.posts.update(post_values)
-      redirect_to community_post_path(@post)
+    @post = Post.find(params[:id])
+    @post.community_id = params[:community_id]
+    if @post.update(post_values)
+      if params[:commit] == "Publish"
+        @post.is_drafted = false
+        @post.save
+        redirect_to community_path(@post.community_id)
+      else 
+        @post.is_drafted = true
+        redirect_to draft_path
+      end  
```

Comment:
This is appearing again. We fixed it already. Who messed this up? @mithun30052001 @vasanthakumar-a

Reply:
I didn't do any resolve conflicts after that magesh